lighthouse-core/audits/seo/tap-targets.js

```diff
@@ -0,0 +1,334 @@
+    );
+
+    if (!symmetricFailure) {
+      failuresAfterMerging.push(failure);
+      return;
+    }
+
+    const {overlapScoreRatio: failureOSR} = failure;
+    const {overlapScoreRatio: symmetricOSR} = symmetricFailure;
+    // Push if:
+    // - the current failure has a higher OSR
+    // - OSRs are the same, and the current failure comes before its symmetric partner in the list
+    // Otherwise do nothing and let the symmetric partner be pushed later.
+    if (failureOSR > symmetricOSR || (
+      failureOSR === symmetricOSR &&
+      overlapFailureIndex < overlapFailures.indexOf(symmetricFailure)
+    )) {
+      failuresAfterMerging.push(failure);
+    }
+  });
+
+  return failuresAfterMerging;
+}
+
+/**
+ * @param {TapTargetOverlapFailure[]} overlapFailures
+ * @returns {LH.Audit.TapTargetTableItem[]}
+ */
+function getTableItems(overlapFailures) {
+  const tableItems = overlapFailures.map(
+    ({
+      tapTarget,
+      overlappingTarget,
+      tapTargetScore,
+      overlappingTargetScore,
+      overlapScoreRatio,
+    }) => {
+      const largestCR = getLargestRect(tapTarget.clientRects);
+      const width = Math.floor(largestCR.width);
+      const height = Math.floor(largestCR.height);
+      const size = width + 'x' + height;
+      return {
+        tapTarget: targetToTableNode(tapTarget),
+        overlappingTarget: targetToTableNode(overlappingTarget),
+        size,
+        width,
+        height,
+        tapTargetScore,
+        overlappingTargetScore,
+        overlapScoreRatio,
+      };
+    });
+
+  tableItems.sort((a, b) => {
+    return b.overlapScoreRatio - a.overlapScoreRatio;
+  });
+
+  return tableItems;
+}
+
+/**
+ * @param {LH.Artifacts.TapTarget} target
+ * @returns {LH.Audit.DetailsRendererNodeDetailsJSON}
+ */
+function targetToTableNode(target) {
+  return {
+    type: 'node',
+    snippet: target.snippet,
+    path: target.path,
+    selector: target.selector,
+  };
+}
+
+class TapTargets extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'tap-targets',
+      title: 'Tap targets are sized appropriately',
+      failureTitle: 'Tap targets are not sized appropriately',
+      description:
+        'Interactive elements like buttons and links should be large enough (48x48px), and have enough space around them, to be easy enough to tap without overlapping onto other elements. [Learn more](https://developers.google.com/web/fundamentals/accessibility/accessible-styles#multi-device_responsive_design).',
+      requiredArtifacts: ['MetaElements', 'TapTargets'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
+   */
+  static audit(artifacts) {
+    const hasViewportSet = ViewportAudit.audit(artifacts).rawValue;
+    if (!hasViewportSet) {
+      return {
+        rawValue: false,
+        explanation:
+          'Tap targets are too small because of a missing viewport config',
```

Comment:
Yeah it should fail if the viewport tag doesn't make it work for mobile. What do you think of this explanation:

> Tap targets are too small because there's no viewport meta tag optimized for mobile screens

Changed it here and in the font size audit.

Reply:
> What do you think of this explanation

sounds great!